pgoapi/pgoapi.py

```diff
@@ -380,8 +384,33 @@ def cleanup_pokemon(self, inventory_items=None):
             if len(pokemons) > MIN_SIMILAR_POKEMON:
                 pokemons = sorted(pokemons, key=lambda x: (x.iv, x.cp), reverse=True)
                 # keep the first pokemon....
+                last_pokemon = pokemons[0]
                 for pokemon in pokemons[MIN_SIMILAR_POKEMON:]:
-                    if pokemon.iv < self.MIN_KEEP_IV and pokemon.cp < self.KEEP_CP_OVER and pokemon.is_valid_pokemon():
+                    if self.pokemon_lvl(pokemon) > self.pokemon_lvl(last_pokemon) and self.RELEASE_DUPLICATES:
+                        if self.pokemon_lvl(pokemon) * self.RELEASE_DUPLICATES_SCALER > self.pokemon_lvl(last_pokemon) and self.pokemon_lvl(last_pokemon) < self.RELEASE_DUPLICATES_MAX_LV:
+                            self.log.info("Releasing pokemon: %s", last_pokemon)
+                            self.release_pokemon(pokemon_id = last_pokemon.id)
+                            release_res = self.call()['responses']['RELEASE_POKEMON']
+                            status = release_res.get('result', -1)
+                            if status == 1:
+                                self.log.info("Successfully Released Pokemon %s", pokemon)
+                            else:
+                                self.log.debug("Failed to release pokemon %s, %s", pokemon, release_res)
+                                self.log.info("Failed to release Pokemon %s", pokemon)
+                            sleep(3)
+                        last_pokemon = pokemon
+                    elif self.pokemon_lvl(last_pokemon) * self.RELEASE_DUPLICATES_SCALER > self.pokemon_lvl(pokemon) and self.pokemon_lvl(pokemon) < self.RELEASE_DUPLICATES_MAX_LV and self.RELEASE_DUPLICATES:
+                        self.log.info("Releasing pokemon: %s", pokemon)
+                        self.release_pokemon(pokemon_id = pokemon.id)
+                        release_res = self.call()['responses']['RELEASE_POKEMON']
+                        status = release_res.get('result', -1)
+                        if status == 1:
+                            self.log.info("Successfully Released Pokemon %s", pokemon)
+                        else:
+                            self.log.debug("Failed to release pokemon %s, %s", pokemon, release_res)
+                            self.log.info("Failed to release Pokemon %s", pokemon)
+                        sleep(3)
+                    elif pokemon.iv < self.MIN_KEEP_IV and pokemon.cp < self.KEEP_CP_OVER and pokemon.is_valid_pokemon():
```

Comment:
I don't think the logic is working correctly, it always goes through this elif (line 413), never through the other ones, and ends up throwing the higher CP ones sometimes.